lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -0,0 +1,43 @@
+module CC
+  module Engine
+    module Analyzers
+      class AnalyzerBase
```

Comment:
I was thinking about it but marginally decided against it. I'll change it to be just `Base`.